kube-runtime/src/watcher.rs

```diff
@@ -331,6 +331,26 @@ impl Config {
         self
     }
 
+    /// Configure typed label selectors
+    ///
+    /// Configure typed selectors from [`Selector`](kube_client::core::Selector) and [`Expression`](kube_client::core::Expression) lists.
+    ///
+    /// ```
+    /// use kube_runtime::watcher::Config;
+    /// use kube_client::core::{Expression, Selector};
+    /// use k8s_openapi::apimachinery::pkg::apis::meta::v1::LabelSelector;
+    /// let selector: Selector = Expression::In("env".into(), ["development".into(), "sandbox".into()].into()).into();
+    /// let cfg = Config::default().labels_from(&selector);
+    /// let cfg = Config::default().labels_from(&Expression::Exists("foo".into()).into());
+    /// // Alternatively the raw LabelSelector is accepted
+    /// let cfg = Config::default().labels_from(&LabelSelector::default().into());
+    ///```
+    #[must_use]
+    pub fn labels_from(mut self, selector: &Selector) -> Self {
+        self.label_selector = Some(selector.to_string());
+        self
+    }
```

Comment:
Added some within the big one.

Reply:
they are good, it's just think it's useful to have some surface level tests outside the core module (like here to check that the `to_string` call is actually using the right underlying serialisation mechanic) as a sanity

it could be something simple like:

```rust
    #[test]
    fn list_params_serialize() {
        let selector: Selector = Expression::In("env".into(), ["development".into(), "sandbox".into()].into()).into();
        let lp = ListParams::default().labels_from(&selector);
        let labels = lp.label_selector.unwrap();
        assert_eq!(labels, "env in (development, sandbox)");
    }
```

Reply:
Added some to the `params.rs`